### ISSUE TITLE
How to transfer a token 

### ISSUE THREAD

**Issue by mzfr** [input-output-hk/cardano-clusterlib-py#56]
I wanted to ask how would I transfer a token from 1 address to another.

```python
destination = [clusterlib.TxOut(address=dst_addr, amount=amount, coin=coin)]
tx_files = clusterlib.TxFiles(signing_key_files=[WALLET_SKEY])

tx_raw_output = cluster.send_funds(
        src_address=SRC_ADDR,
        destinations=destination,
        tx_name="send_funds",
        tx_files=tx_files,
    )
```
Here amount was `1800000`, coin was `<POLICY_ID>.<name>`(`0c068e25a5469612d82b256cf9feac9343bb98dd91c7f636c58f17ec.Testtoken`), dst_addr was a testnet address

I understand that there is `send_tx` but I couldn't figure out how to use that either.

**Comment by mzfr** [comment]
@mkoura Any suggestions on this?

**Comment by mkoura** [comment]
What error do you get? It is probably related to the fact that you can't create a UTxO without any Lovelace in it. So together with the token you also need to transfer some minimum Lovelace amount:

```python
destinations = [clusterlib.TxOut(address=dst_addr, amount=amount, coin=coin), clusterlib.TxOut(address=dst_addr, amount=1500_000)]
...
```

Also current node master expect asset names to be hex encoded, e.g.:
```python
asset_name_enc = asset_name.encode("utf-8").hex()
coin = f"{policyid}.{asset_name_enc}"
```
This change is not deployed to testnet yet, so your coin `0c068e25a5469612d82b256cf9feac9343bb98dd91c7f636c58f17ec.Testtoken` is correct for now.

**Comment by mzfr** [comment]
This is the error that I got

```
('Invalid Address', CLIError('An error occurred running a CLI command `cardano-cli transaction submit --testnet-magic 1097911063 --tx-file send_funds_tx.signed --cardano-mode` on path `/home/transaction_api`: Command failed: transaction submit  Error: Error while submitting tx: ShelleyTxValidationError ShelleyBasedEraAlonzo (ApplyTxError [UtxowFailure (WrappedShelleyEraFailure (UtxoFailure (ValueNotConservedUTxO (Value 997416407 (fromList [(PolicyID {policyID = ScriptHash "6b8d07d69639e9413dd637a1a815a7323c69c86abbafb66dbfdb1aa7"},fromList [("",2)])])) (Value 997416407 (fromList [(PolicyID {policyID = ScriptHash "6b8d07d69639e9413dd637a1a815a7323c69c86abbafb66dbfdb1aa7"},fromList [("",1800000)])])))))])\n'))
```

I'm doing the transaction in the following way:

```python
amount = 1800000

destination = [clusterlib.TxOut(address=dst_addr, amount=amount, coin="6b8d07d69639e9413dd637a1a815a7323c69c86abbafb66dbfdb1aa7"), clusterlib.TxOut(address=dst_addr, amount=1500000)]

tx_files = clusterlib.TxFiles(signing_key_files=[sign_key])

 tx_raw_output = CLI.send_funds(
                src_address=own_addr,
                destinations=destination,
                tx_name="send_funds",
                tx_files=tx_files,
            )
```

Also to give you the view in the testnet wallet I have the following utxo:

```
                           TxHash                                 TxIx        Amount
--------------------------------------------------------------------------------------
a23d5490d5efb77ed1f65e1a8006a02df11690f36924fc20d41e118ab6533f25     1        996106091 lovelace + TxOutDatumHashNone
f91b0f77dbec60e01b5f64ec2ef6df4d8be66220a131ad2d6df12c0e8e41122e     0        1310316 lovelace + 2 6b8d07d69639e9413dd637a1a815a7323c69c86abbafb66dbfdb1aa7 + TxOutDatumHashNone

```

It does say `value not conserved`. Not sure whether it's because I have 2 tokens or something else. I tried to pass the coin argument in the following manner but still got error:

```python
coin="1 6b8d07d69639e9413dd637a1a815a7323c69c86abbafb66dbfdb1aa7"
```

**Comment by mkoura** [comment]
Looks like the amount of token `6b8d07d69639e9413dd637a1a815a7323c69c86abbafb66dbfdb1aa7` that you have available is 2, and you are trying to transfer 1800000 of them.

**Comment by mzfr** [comment]
@mkoura Yes that was the issue. Thanks for all the help :)
